bluesky-social/social-app
# Fabric migration - `rn-fetch-blob`, `react-native-fs` -> `expo-file-system`

### Testing requires building a client after checking out https://github.com/bluesky-social/social-app/pull/3096

This removes migrates our use of `react-native-fs` and `rn-fetch-blob` to `expo-file-system`. It also makes use of `expo-image-manipulator` instead of using `react-native-image-resizer`.

It also fixes a bug where the image manipulator would try to save to the same file path again. We are now cleaning up after ourselves, so this can no longer happen. Of course, in cleaning up we are also preventing the unnecessary growth of the local file store as well.

Tests are also updated to reflect the use of new libraries.

## Testing

- Save image (Tests saving and cache)
	- [x] Not applicable on web
- Share image (Tests saving and cache)
	- [x] Not applicable on web
- Add a link card (Tests both saving and resizing/compression)
	- [x] Web - No changes, but verify it still works
- Upload an image (Tests resizing/compression) 
	- [x] Web - No changes, but verify it still works
- Upload an avatar (Tests resizing/compression)
	- [x] Web - No changes, but verify it still works
- Upload a banner (Tests resizing/compression)
	- [x] Web - No changes, but verify it still works
